mlflow/models/utils.py

```diff
@@ -2019,3 +2021,35 @@ def validate_serving_input(model_uri: str, serving_input: Union[str, dict[str, A
     finally:
         if output_dir and os.path.exists(output_dir):
             shutil.rmtree(output_dir)
+
+
+def get_external_mlflow_model_spec(logged_model: LoggedModel) -> Model:
+    """
+    Create the MLflow Model specification for a given logged model whose artifacts
+    (code, weights, etc.) are stored externally outside of MLflow.
+
+    Args:
+        logged_model: The external logged model for which to create an MLflow Model specification.
+
+    Returns:
+        Model: MLflow Model specification for the given logged model with external artifacts.
+    """
+    from mlflow.models.signature import infer_signature
+
+    return Model(
+        artifact_path=logged_model.artifact_location,
+        model_uuid=logged_model.model_id,
+        model_id=logged_model.model_id,
+        run_id=logged_model.source_run_id,
+        # Include a dummy signature so that the model can be registered to the Databricks Unity
+        # Catalog Model Registry.
+        # TODO: Remove this once the Databricks Unity Catalog Model Registry supports registration
+        # of models without signatures
+        signature=infer_signature(model_input=True, model_output=True),
```

Comment:
I've confirmed that this works in the Databricks UC Model Registry. Models created with `external=True` can be registered.